File: library/src/main/java/me/yamlee/jsbridge/utils/InputTypeUtil.java
package me.yamlee.jsbridge.utils;

import android.app.Activity;
import android.content.ClipData;
import android.content.Context;
import android.os.Build;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import timber.log.Timber;

/**
 * 输入法工具类,如:打开关闭软键盘,复制,粘帖文字等
 *
 * @author yamlee
 */
public class InputTypeUtil {
    @SuppressWarnings("HardCodedStringLiteral")
    private static final String TAG = "InputTypeUtil";

    /**
     * 保存数据到剪切板
     *
     * @param context
     * @param str
     */
    public static void saveClipBoard(Context context, String str) {
        if (context == null) {
            return;
        }

        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {

            android.text.ClipboardManager cmb = (android.text.ClipboardManager) context
                    .getSystemService(Activity.CLIPBOARD_SERVICE);
            cmb.setText(str);
        } else {
            android.content.ClipboardManager clipboard = (android.content.ClipboardManager) context
                    .getSystemService(Activity.CLIPBOARD_SERVICE);
            @SuppressWarnings("HardCodedStringLiteral") ClipData clip = ClipData.newPlainText("honey", str);
            clipboard.setPrimaryClip(clip);
        }
    }

    /**
     * 打开软件盘
     *
     * @param context
     * @param view
     */
    public static void openSoftKeyBoard(Context context, View view) {
        InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        if (view != null) {
            //noinspection HardCodedStringLiteral
            Timber.i("-----打开软键盘------");
            imm.showSoftInput(view, InputMethodManager.SHOW_FORCED);
        }
    }

    /**
     * 关闭软键盘
     *
     * @param context
     * @param view
     */
    public static void closeSoftKeyBoard(Context context, View view) {
        InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        if (view != null) {
            if (view.getWindowToken() != null) {
                imm.hideSoftInputFromWindow(view.getApplicationWindowToken(), 0);
            }
        }
    }


}

File: android-jsBridgeCore/src/main/java/me/yamlee/jsbridge/utils/InputTypeUtil.kt
package me.yamlee.jsbridge.utils

import android.app.Activity
import android.content.ClipData
import android.content.Context
import android.os.Build
import android.view.View
import android.view.inputmethod.InputMethodManager

/**
 * 输入法工具类,如:打开关闭软键盘,复制,粘帖文字等
 *
 * @author yamlee
 */
object InputTypeUtil {
    private val TAG = "InputTypeUtil"

    /**
     * 保存数据到剪切板
     *
     * @param context
     * @param str
     */
    fun saveClipBoard(context: Context?, str: String) {
        if (context == null) {
            return
        }

        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {

            val cmb = context
                    .getSystemService(Activity.CLIPBOARD_SERVICE) as android.text.ClipboardManager
            cmb.text = str
        } else {
            val clipboard = context
                    .getSystemService(Activity.CLIPBOARD_SERVICE) as android.content.ClipboardManager
            val clip = ClipData.newPlainText("honey", str)
            clipboard.primaryClip = clip
        }
    }

    /**
     * 打开软件盘
     *
     * @param context
     * @param view
     */
    fun openSoftKeyBoard(context: Context, view: View?) {
        val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        if (view != null) {

            LogUtil.info("-----open soft key board------")
            imm.showSoftInput(view, InputMethodManager.SHOW_FORCED)
        }
    }

    /**
     * 关闭软键盘
     *
     * @param context
     * @param view
     */
    fun closeSoftKeyBoard(context: Context, view: View?) {
        val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        if (view != null) {
            if (view.windowToken != null) {
                imm.hideSoftInputFromWindow(view.applicationWindowToken, 0)
            }
        }
    }


}
